src/main/java/org/testng/internal/PackageUtils.java

```diff
@@ -111,9 +111,9 @@ else if ("jar".equals(protocol)) {
               packageName = name.substring(0, idx).replace('/', '.');
             }
 
-            Utils.log("PackageUtils", 4, "Package name is " + packageName);
-            if ((idx != -1) || recursive) {
+            if (recursive || packageName.equals(packageOnly)) {
               //it's not inside a deeper dir
+              Utils.log("PackageUtils", 4, "Package name is " + packageName);
```

Comment:
We should log like in 'file' protocol
https://github.com/cbeust/testng/pull/618/files#diff-e366fcca04f9d843a654346f0e7255e4R235

only if dir/package was  succefully approved for scaning (https://github.com/cbeust/testng/pull/618/files#diff-e366fcca04f9d843a654346f0e7255e4R228)